### ISSUE TITLE
asdf and running tests

### ISSUE THREAD

**Issue by Linuus** [aaronjensen/spacemacs.d#5]
Hi there!

Sorry to bother you :)

I'm trying to get asdf and zsh to work properly with spacemacs and I can see that you have both of them as well!

My issue is that if I pop open a shell in emacs (eshell, ansi-term or whatever) I can run, for instance, my ruby tests just fine with `bundle exec rspec`.
However, when I do something like running a single test directly with `SPC t t` it just errors out saying it can't find command `bundle` or whatever, which makes me think this is a PATH issue.

So, I'm just curious if this scenario works for you? Have you seen the same issue I have?

Cheers!

Linus

**Comment by aaronjensen** [comment]
Hey @Linuus, no worries.

You may be running into the same thing that I ran into, which I solved with this: https://github.com/aaronjensen/dotfiles/commit/8ce2ef722d1dadead55861a3f3299ef934f28765

If you're using bash, you'll probably need to do something similar. Basically, you need to ensure that non-interactive shells get asdf sourced.

**Comment by Linuus** [comment]
Yeah, I did that when I saw your config but it didn't help. I put `echo "ZSHENV"` in my zhenv file, `echo "ZSHRC"` in zshrc etc. and figured out this:

- When opening a new shell in iTerm I get "ZSHENV, ZPROFILE, ZSHRC"
- When opening a new shell in emacs (ansi-term for instance) I get "ZSHENV, ZSHRC"
- When running tests with `SPC m t t` I get "ZSHENV".

So, now I'm just sourcing asdf if .zshenv instead and it works everywhere it seems. The weird thing is that I did that before, but something didn't work (I can't remember what of course... 😞 ). Perhaps I'll find out soon enough 😄 

Thank you for your time!
